apis/field_error.go

```diff
@@ -312,20 +312,29 @@ func ErrDisallowedUpdateDeprecatedFields(fieldPaths ...string) *FieldError {
 }
 
 // ErrInvalidArrayValue constructs a FieldError for a repetetive `field`
-// at `index` that has received an invalid string value.
+// at `index` that has received an invalid value.
 func ErrInvalidArrayValue(value interface{}, field string, index int) *FieldError {
 	return ErrInvalidValue(value, CurrentField).ViaFieldIndex(field, index)
 }
 
 // ErrInvalidValue constructs a FieldError for a field that has received an
-// invalid string value.
+// invalid value.
 func ErrInvalidValue(value interface{}, fieldPath string) *FieldError {
 	return &FieldError{
 		Message: fmt.Sprintf("invalid value: %v", value),
 		Paths:   []string{fieldPath},
 	}
 }
 
+// ErrInvalidCombination constructs a FieldError for a field that has received an
+// invalid combination of values.
+func ErrInvalidCombination(bad, value interface{}, key, fieldPath string) *FieldError {
+	return &FieldError{
+		Message: fmt.Sprintf("invalid value %v, when %s is %v", bad, key, value),
```

Comment:
This could be confusing as `failureThreshold` is a key and not a value. To move towards Evan's suggestion, perhaps rephrase as: "%s must be unset when %s is %v"

For example,
`failureThreshold` must be unset when `periodSeconds` is `zero` <-- Works for current validation errors
`foo` must be unset when `bar` is `present` <-- Works for strict A or B errors where value does not matter

Reply:
No, I wanted to use the arg `bad` as a value, not key. Invalid value's key could be printed by [`Paths: []string{fieldPath}`](https://github.com/knative/pkg/pull/638/files#diff-d4c2f4719b228e9168c970edd02d87a3R334).